### ISSUE TITLE
Formatting a function that returns a struct with a single element with the last field commented out will cause code to not compile

### ISSUE THREAD

**Issue by werecat** [rust-lang-nursery/rustfmt#804]
If you have code that looks like this
```rust
fn new() -> SomeStruct {
    SomeStruct {
        field_a: 5,
        // field_b: 3,
    }
}
```

Then rustfmt will change it to this, which would not compile due to missing delimiters
```rust
fn new() -> SomeStruct {
    SomeStruct { field_a: 5, // field_b: 3, }
}
```

This only seems to happen when `field_a: 5` has a `,` at the end and when field_a is the only field in the struct. Otherwise rustfmt changes the single line comment into an enclosed multiline comment, which does not break code.

rustfmt should correctly change the original code at the top to
```rust
fn new() -> SomeStruct {
    SomeStruct { field_a: 5, /* field_b: 3, */ }
}
```
Or perhaps
```rust
fn new() -> SomeStruct {
    SomeStruct { field_a: 5 /* field_b: 3, */ }
}
```

**Comment by DarkDrek** [comment]
This doesn't seem to happen on the latest version from the repo and I could trace the fix back to the merge of #756. After the PR we don't generate wrong code. So this could be fixed with a new version to creates.io  #805.

**Comment by nrc** [comment]
Thanks for investigating! I intend to make a new release pretty soon, so I'll close this issue for now.
